server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -156,9 +164,19 @@ private void handleUpdate(MailboxId mailboxId, MailboxUpdateRequest updateReques
         }
    }
 
-    private void checkRole(Optional<Role> role) throws SystemMailboxNotUpdatableException {
-        if (role.map(Role::isSystemRole).orElse(false)) {
-            throw new SystemMailboxNotUpdatableException();
+    private void assertNotSharedOutboxOrDraftMailbox(Mailbox mailbox, MailboxUpdateRequest updateRequest) {
+        Preconditions.checkArgument(!updateRequest.getSharedWith().isPresent() || !mailbox.hasRole(Role.OUTBOX), "The system does not allow sharing 'Outbox'");
+        Preconditions.checkArgument(!updateRequest.getSharedWith().isPresent() || !mailbox.hasRole(Role.DRAFTS), "The system does not allow sharing 'Draft'");
+    }
+
+    private void assertSystemMailboxesAreNotUpdated(Mailbox mailbox, MailboxUpdateRequest updateRequest) throws SystemMailboxNotUpdatableException {
```

Comment:
I don't see where this code is tested (and BTW I don't understand what it does)

Reply:
The old code said, it does not allow to update system mailbox like: change mailbox name, mailbox role or ... And it had integration test before like setMailboxesShouldReturnNotUpdatedWhenRenamingSystemMailbox